apps/studio/src/server/modules/user/user.router.ts

```diff
@@ -0,0 +1,282 @@
+import { TRPCError } from "@trpc/server"
+
+import {
+  countInputSchema,
+  countOutputSchema,
+  createInputSchema,
+  createOutputSchema,
+  deleteInputSchema,
+  deleteOutputSchema,
+  getUserInputSchema,
+  getUserOutputSchema,
+  hasInactiveUsersInputSchema,
+  hasInactiveUsersOutputSchema,
+  listInputSchema,
+  listOutputSchema,
+  updateInputSchema,
+  updateOutputSchema,
+} from "~/schemas/user"
+import { protectedProcedure, router } from "../../trpc"
+import { db, sql } from "../database"
+import { validatePermissionsForManagingUsers } from "../permissions/permissions.service"
+import {
+  createUser,
+  getUsersQuery,
+  validateEmailRoleCombination,
+} from "./user.service"
+
+export const userRouter = router({
+  create: protectedProcedure
+    .input(createInputSchema)
+    .output(createOutputSchema)
+    .mutation(async ({ ctx, input: { siteId, users } }) => {
+      await validatePermissionsForManagingUsers({
+        siteId,
+        userId: ctx.user.id,
+        action: "create",
+      })
+
+      const createdUsers = await db.transaction().execute(async (trx) => {
+        return await Promise.all(
+          users.map(async (user) => {
+            const { user: createdUser, resourcePermission } = await createUser({
+              ...user,
+              email: user.email.toLowerCase(),
+              siteId,
+              trx,
+            })
+            return {
+              id: createdUser.id,
+              email: createdUser.email,
+              role: resourcePermission.role,
+            }
+          }),
```

Comment:
[issue] this feels like an issue because we're holding a lock and inserting row after row into db. wonder if we should use a bulk insert here? (see kysely's [docs](https://kysely.dev/docs/examples/insert/multiple-rows))